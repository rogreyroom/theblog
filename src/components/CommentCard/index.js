import React from 'react'
import PropTypes from 'prop-types'
import classes from './index.module.scss'

const CommentCard = ({ name, body }) => (
  <section className={classes.card}>
    <h4 className={classes.card__title}>{name}</h4>
    <p className={classes.card__text}>{body}</p>
  </section>
)

CommentCard.propTypes = {
  name: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
}

export default CommentCard
