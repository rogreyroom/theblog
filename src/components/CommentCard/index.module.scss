.card {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  border: solid 3px var(--c-accent);

  &__title {
    color: var(--c-black);
    font-size: clamp(var(--fs-text-normal), 0.7212rem + 0.7692vw,  var(--fs-text-large));
    display: block;
    width: 100%;
    background-color: var(--c-accent-alpha);
    margin: 0;
    padding: clamp(var(--xs), 0.2241rem + 1.3793vw, var(--normal));
  }

  &__text {
    color: var(--c-black);
    font-size: clamp(var(--fs-text-small), 0.7212rem + 0.7692vw,  var(--fs-text-large));
    margin: 0;
    display: block;
    width: 100%;
    padding: clamp(var(--xs), 0.2241rem + 1.3793vw, var(--normal));
  }
}
