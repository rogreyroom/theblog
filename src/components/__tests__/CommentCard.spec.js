/**
 * @jest-environment jsdom
 */
import React from 'react'
import { screen, render, cleanup } from '@testing-library/react'
import CommentCard from '../CommentCard'

describe('CommentCard component', () => {
  let element
  beforeEach(() => {
    const { container } = render(
      <CommentCard name="John" body="Some nice comment about the post" />,
    )
    element = container
  })

  it('renders the component', () => {
    expect(element).toMatchSnapshot()
  })

  it('component contains text "John"', () => {
    expect(screen.getByText(/john/i)).toHaveTextContent('John')
  })

  it('component contains text "Some nice comment about the post"', () => {
    expect(screen.getByText(/comment/i)).toHaveTextContent(
      'Some nice comment about the post',
    )
  })

  afterAll(cleanup)
})
