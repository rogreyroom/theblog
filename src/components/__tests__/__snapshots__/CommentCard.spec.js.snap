// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard component renders the component 1`] = `
<div>
  <section
    class="card"
  >
    <h4
      class="card__title"
    >
      John
    </h4>
    <p
      class="card__text"
    >
      Some nice comment about the post
    </p>
  </section>
</div>
`;
